### PR TITLE
Updated validating VS toolset generic for different VS versions

### PR DESCRIPTION
**Pull request by jainaashish**
Currently there are multiple places in our common.ps1 or configure.ps1 which only uses dev15 specific version or env variables which is why it fails for dev16 at some machines. This PR will fix those instances to make it generic for any VS version.

### CODE REVIEW

File: configure.ps1 [hunk 1]
```diff
@@ -129,20 +129,23 @@ if (-not $ProgramFiles -or -not (Test-Path $ProgramFiles)) {
 $MSBuildDefaultRoot = Get-MSBuildRoot
 $MSBuildRelativePath = 'bin\msbuild.exe'
 
-Invoke-BuildStep 'Validating VS15 toolset installation' {
-    $vs15 = New-BuildToolset 15
+Invoke-BuildStep 'Validating VS toolset installation' {
+
+    $vsMajorVersion = Get-VSMajorVersion
```
Comment (nkolev92) [inline configure.ps1:134]: I think there's another case that can be handled. 

Say you only have VS16 on your machine, and you are not running this in developer command prompt. 
 
You'd need to pass special switches to force configure.ps1 to pick it up. 

Ideally when discovering the VS installation, you can return a tuple of (version, msbuildpath). 
Then based on which version was discovered here, validate the toolset.

File: build/common.ps1 [hunk 2]
```diff
@@ -315,16 +317,34 @@ Function Get-LatestVisualStudioRoot() {
...
     )
+
+    $vsMajorVersion = Get-VSMajorVersion
+
     # Willow install workaround
     if (-not $Default) {
+
         # Find version 15.0 or newer
-        if (Test-Path Env:\VS150COMNTOOLS) {
+        $CommonToolsVar = "Env:VS${vsMajorVersion}0COMNTOOLS"
```
Comment (nkolev92) [inline build/common.ps1:344]: Personally I'd let this here determine the version. 
Look for either 15/16 version of the env variable. 
If either exists use it.

**Review by nkolev92 — COMMENTED** [review]
Some suggestions. 
Curious what you think about that. 
We can discuss it offline if easier.

File: configure.ps1 (hunk 1 above)
Comment (jainaashish) [reply configure.ps1:134]: >Say you only have VS16 on your machine, and you are not running this in developer command prompt.

This will work in that case as well because while building toolset object, we also use vswhere to find the latest VS installation and use that. Corresponding code at https://github.com/NuGet/NuGet.Client/blob/dev/configure.ps1#L94

File: build/common.ps1 (hunk 2 above)
Comment (jainaashish) [reply build/common.ps1:344]: There are multiple reasons not to do that.
1) it will not be extensible for future VS versions. You will keep adding more and more VS versions manually
2) If `VisualStudioVersion` env is not set in current prompt then its highly unlikely that `VS160COMNTOOLS` or `VS150COMNTOOLS` would be there so at that point in time, this will always go to the fallback to use VSWHERE to pick the latest VS installation.
3) We're using many ENV variables with fallbacks which seems right to me.

Happy to discuss in person if that helps to understand

Comment (nkolev92) [reply build/common.ps1:344]: As currently I'm not sure if the following scenario is handled. 
I have a:
- Dev16 Preview
- Dev16 stable

The preview is the higher version. 

I am in the Dev16 stable developer command prompt, and I want to build using that same stable msbuild version (we've had cases where the previews have bugs that prevent us from building). 

Right now I don't see that handled, as Dev15 is the default. 

In Dev15, this would have been autopicked. 

tldr; 

I think if you are in a valid compatbile developer command prompt (Dev15/Dev16), you should just build with that. 

Let's chat offline.

Comment (nkolev92) [reply build/common.ps1:344]: This is handled by Get-VSVersion. 
Made a small change in the retrieval of the env var to make this more consistent.

**Review by nkolev92 — APPROVED** [review]
Looks and works great.

**Comment by nkolev92** [comment]
Well, at least locally it does, the CI is having issues, so we'll hold off merging for now.

**Comment by jainaashish** [comment]
There was a minor issue with `Split` method. Fixed and merged the PR.
